tests/views/get_edit_unit.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import json
+
+import pytest
+
+from django.db.models import Max
+
+from pootle_app.models.permissions import check_user_permission
+from pootle_store.models import Unit
+from pootle_store.util import find_altsrcs
+from pootle_store.views import get_alt_src_langs
+
+
+@pytest.mark.django_db
+def test_get_edit_unit(client, request_users, settings):
+
+    user = request_users["user"]
+    if user.username != "nobody":
+        client.login(
+            username=user.username,
+            password=request_users["password"])
+
+    unit = Unit.objects.get_translatable(user).first()
+    store = unit.store
+    directory = store.parent
+    translation_project = store.translation_project
+    project = translation_project.project
+    language = translation_project.language
+
+    response = client.get(
+        "/xhr/units/%s/edit/" % unit.id,
+        HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+    request = response.wsgi_request
+    result = json.loads(response.content)
+
+    priority = None
+
+    if 'virtualfolder' in settings.INSTALLED_APPS:
+        vfolder_pk = request.GET.get('vfolder', '')
+
+        if vfolder_pk:
+            from virtualfolder.models import VirtualFolder
+
+            try:
+                # If we are translating a virtual folder, then display its
+                # priority.
+                # Note that the passed virtual folder pk might be invalid.
+                priority = VirtualFolder.objects.get(pk=vfolder_pk).priority
+            except VirtualFolder.DoesNotExist:
+                pass
+
+        if priority is None:
+            # Retrieve the unit top priority, if any. This can happen if we are
+            # not in a virtual folder or if the passed virtual folder pk is
+            # invalid.
+            priority = unit.vfolders.aggregate(
+                priority=Max('priority')
+            )['priority']
```

Comment:
If I am not wrong, the priority when no virtualfolders exist for this unit is `1`, so it is safe to default to it instead of `None` and get rid of this `if`.